From 9dc8f26d9857ffe20e243566b74fa4b25adabd6a Mon Sep 17 00:00:00 2001
From: Davide Beatrici <git@davidebeatrici.dev>
Date: Sat, 13 Aug 2022 14:24:45 +0200
Subject: [PATCH] build system: Allow using stb from the system.

---
 src/CMakeLists.txt | 19 ++++++++++++++-----
 1 file changed, 14 insertions(+), 5 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 265cc53..6f0fc02 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -170,11 +170,20 @@ else() # Other compiler/version are unsupported, use external lib
 endif()
 
 ## stb ##
-execute_process(
-	COMMAND git submodule update --init src/stb
-	WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
-)
-include_directories(${CMAKE_CURRENT_SOURCE_DIR}/stb)
+if(NOT WIN32 OR MINGW)
+	pkg_check_modules(STB REQUIRED stb)
+endif()
+
+if(STB_FOUND)
+	add_definitions(${STB_CFLAGS_OTHER})
+	include_directories(${STB_INCLUDE_DIRS})
+else()
+	execute_process(
+		COMMAND git submodule update --init src/stb
+		WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+	)
+	include_directories(${CMAKE_CURRENT_SOURCE_DIR}/stb)
+endif()
 
 ## OpenBoardView ##
 add_subdirectory(openboardview)
-- 
2.35.1

