--- src/CMakeLists.txt
+++ src/CMakeLists.txt
@@ -53,19 +53,7 @@
 endif()
 
 ## glad ##
-execute_process(
-	COMMAND git submodule update --init src/glad
-	WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
-)
-set(GLAD_SOURCES_DIR "${PROJECT_SOURCE_DIR}/src/glad")
-add_subdirectory(glad/cmake glad_cmake)
-
-if(ENABLE_GL1 OR ENABLE_GL3)
-glad_add_library(glad_gl REPRODUCIBLE API gl:compatibility=3.2 EXTENSIONS NONE)
-endif()
-if(ENABLE_GLES2)
-glad_add_library(glad_gles REPRODUCIBLE API gles2=2.0 EXTENSIONS GL_OES_element_index_uint GL_EXT_texture_format_BGRA8888)
-endif()
+find_package(glad REQUIRED)
 
 ## SDL2 ##
 find_package(SDL2 REQUIRED CONFIG)
@@ -142,15 +130,9 @@
 	${CMAKE_CURRENT_SOURCE_DIR}/imgui/examples
 	${CMAKE_CURRENT_SOURCE_DIR}/imgui/misc/cpp
 )
-if(ENABLE_GLES2)
 target_link_libraries(imgui
-  glad_gles
+	glad::glad
 )
-elseif(ENABLE_GL1 OR ENABLE_GL3)
-  target_link_libraries(imgui
-    glad_gl
-  )
-endif()
 if(MINGW)
 target_link_libraries(imgui
 	SDL2::SDL2-static
